File: python/1893.check-if-all-the-integers-in-a-range-are-covered.py
class Solution:
    def isCovered(self, ranges: list[list[int]], left: int, right: int) -> bool:

        cover = [0] * 52
        for start, end in ranges:
            cover[start] += 1
            cover[end + 1] -= 1

        coverage = 0
        for curr in range(1, right + 1):
            coverage += cover[curr]

            if left <= curr and coverage == 0:
                return False

        return True

File: golang/1893.check-if-all-the-integers-in-a-range-are-covered.go
func isCovered(ranges [][]int, left int, right int) bool {

	cover := make([]int, 52)
	for _, r := range ranges {
		cover[r[0]]++
		cover[r[1]+1]--
	}

	coverage := 0

	for i := 1; i <= right; i++ {
		coverage += cover[i]

		if left <= i && coverage == 0 {
			return false
		}
	}

	return true
}
